File: 程式語言 V2答案檔/C++/402.cpp
#include <iostream>
#include <cstring>
using namespace std;

int main()
{
    char str1[128], str2[128];
    int n, result;
    cin.getline(str1, 128);
    cin.getline(str2, 128);
    cin >> n;
    if (n > (int)strlen(str1) || n > (int)strlen(str2)) {
        cout << "error";
    }
    else {
        result = strncmp(str1, str2, n);
        if (result > 0) {
            cout << str1 << " > " << str2;
        }
        else if (result < 0) {
            cout << str1 << " < " << str2;
        }
        else {
            cout << str1 << " = " << str2;
        }
    }
    return (0);
}

File: 程式語言 V2答案檔/C#/402.cs
using System;

public class TQC402
{
    public static void Main(string[] args)
    {
        string str1, str2;
        int n;
        str1 = Console.ReadLine();
        str2 = Console.ReadLine();
        n = int.Parse(Console.ReadLine());

        if (n > str1.Length || n > str2.Length) {
            Console.WriteLine("error");    
        }
        else {
            string subString1 = str1.Substring(0, n);
            string subString2 = str2.Substring(0, n);
            int compare = strncmp(str1, str2, n);
            if (compare == -1)
                Console.WriteLine(str1 + " < " + str2);
            else if (compare == 0)
                Console.WriteLine(str1 + " = " + str2);
            else if (compare == 1)
                Console.WriteLine(str1 + " > " + str2);
        }
    }

    public static int strncmp(string str1, string str2, int n) {
        for (int i = 0; i < n; i++) {
            if (str1[i] < str2[i])
                return -1;
            else if (str1[i] > str2[i])
                return 1;
        }
        return 0;
    }
}